examples/vision/simsiam.py

```diff
@@ -0,0 +1,435 @@
+    else:
+        return x
+
+
+def custom_augment(image):
+    # As discussed in the SimCLR paper, the series of augmentation
+    # transformations (except for random crops) need to be applied
+    # randomly to impose translational invariance.
+    image = flip_random_crop(image)
+    image = random_apply(color_jitter, image, p=0.8)
+    image = random_apply(color_drop, image, p=0.2)
+    return image
+
+
+"""
+It should be noted that an augmentation pipeline is generally dependent on various
+properties of the dataset we are dealing with. For example, if images in the dataset are
+heavily object-centric then taking random crops with a very high probability may hurt the
+training performance.
+
+Let's now apply our augmentation pipeline to our dataset and visualize a few outputs. 
+"""
+
+"""
+## Convert the data into TensorFlow `Dataset` objects
+
+Here we create two different versions of our dataset *without* any ground-truth labels. 
+"""
+
+ssl_ds_one = tf.data.Dataset.from_tensor_slices(x_train)
+ssl_ds_one = (
+    ssl_ds_one.shuffle(1024, seed=SEED)
+    .map(custom_augment, num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+ssl_ds_two = tf.data.Dataset.from_tensor_slices(x_train)
+ssl_ds_two = (
+    ssl_ds_two.shuffle(1024, seed=SEED)
+    .map(custom_augment, num_parallel_calls=AUTO)
+    .batch(BATCH_SIZE)
+    .prefetch(AUTO)
+)
+
+# We then zip both of these datasets.
+ssl_ds = tf.data.Dataset.zip((ssl_ds_one, ssl_ds_two))
+
+# Visualize a few augmented images.
+sample_images_one = next(iter(ssl_ds_one))
+plt.figure(figsize=(10, 10))
+for n in range(25):
+    ax = plt.subplot(5, 5, n + 1)
+    plt.imshow(sample_images_one[n].numpy().astype("int"))
+    plt.axis("off")
+plt.show()
+
+# Ensure that the different versions of the dataset actually contain
+# identical images.
+sample_images_two = next(iter(ssl_ds_two))
+plt.figure(figsize=(10, 10))
+for n in range(25):
+    ax = plt.subplot(5, 5, n + 1)
+    plt.imshow(sample_images_two[n].numpy().astype("int"))
+    plt.axis("off")
+plt.show()
+
+"""
+Notice that the images in `samples_images_one` and `sample_images_two` are essentially
+the same but are augmented differently.
+"""
+
+"""
+## Defining the encoder and the predictor
+
+We use an implementation of ResNet20 that is specifically configured for the CIFAR10
+dataset. The code is taken from the
+[keras-idiomatic-programmer](https://github.com/GoogleCloudPlatform/keras-idiomatic-programmer/blob/master/zoo/resnet/resnet_cifar10_v2.py) repository. The hyperparameters of
+these architectures have been referred from Section 3 and Appendix A of [the original
+paper](https://arxiv.org/abs/2011.10566). 
+"""
+
+"""shell
+!wget -q
+https://gist.githubusercontent.com/sayakpaul/0d2d367a9f9e2dd0941e2779c553c314/raw/609f10c2
+c5fade182191c8b2042c4da9348523d7/resnet_cifar10_v2.py \
+    -O resnet_cifar10_v2.py
+"""
+
+import resnet_cifar10_v2
+
+n = 18
+depth = n * 9 + 2
+N_BLOCKS = ((depth - 2) // 9) - 1
```

Comment:
Style nit: if you go with uppercase for all constnats, let's go with uppercase for `n` and `depth` as well. Also prefer using `num_blocks` to be consistent with naming conventions in the Keras codebase in general.